nix/ovpn.nix

```diff
@@ -0,0 +1,43 @@
+{ config, pkgs, ... }:
+{
+  boot = {
+    kernel.sysctl = {
+      "net.ipv4.ip_forward" = 1;
+    };
+  };
+  networking = {
+    firewall = {
+      enable = true;
+      extraCommands = ''
+        iptables -t nat -A POSTROUTING -s 10.8.0.0/24 -o eno1 -j MASQUERADE
+      '';
+      extraStopCommands = ''
+        iptables -t nat -D POSTROUTING -s 10.8.0.0/24 -o eno1 -j MASQUERADE
+      '';
+      allowedUDPPorts = [ 1195 ]; # openvpn
+      trustedInterfaces = [ "tun1" ];
+    };
+  };
+  services = {
+    openvpn = {
+      servers = {
+        for-phone = {
+          config = ''
+            dev tun1
+            server 10.8.0.0 255.255.255.0
+            port 1195
+            comp-lzo
+            ca /root/easy-rsa/easyrsa3/pki/ca.crt
+            cert /root/easy-rsa/easyrsa3/pki/issued/no-pass.crt
```

Comment:
lots of certs:
easyrsa creates a cert to run its own CA, this one i passprhased. this gets used to make all of the following certs.
openvpn server private/pub key certificate, i didnt pasprhase so i can get auto start
client has public/private key, i didnt passprhase because im lazy
dh params file also created by easyrsa

created by openvpn command:
openvpn secret file for the tls security, used on client and server

by not passphrasing the openvpn server cert, i suppose someone with root access to my machine could easily MITM